pagopa/pn-frontend-e2e
# PN-9449-e-2-e-fix-test-che-falliscono

### Issue Description
All tests lack proper data preparation, potentially causing issues such as invalid notification requirements or the absence of data. To mitigate these problems, a proper (temporary) HTTP configuration file and new background tests need to be set up. While this pull request won't include all HTTP requests in the project, it serves as a solid starting point for their implementation, with adjustments as needed.

In the future, the project might migrate to the `Spring` technology, which will speed up the process of configuring everything related to requests and similar functionalities.

### Changes Made:
- Added `CustomHttpClient` class file, required for sending and receiving HTTP requests/responses.
- Added `RestNotification` class file, which declares all HTTP methods related to `Notification`.
- Added custom exception classes to manage `CustomHttpClient` and `RestNotification`.
- Added custom model classes to manage HTTP response/request.
- Added custom enums.
- Added a new test class file to separate the logic between background tests and real E2E tests.
- Adjusted tests related to downloading attachments.
- Removed some unnecessary comments.

### How to Test
Simply run mvn clean test `-Dcucumber.filter.tags=@TA_MittenteDownloadFileDettaglioNotifica` and verify if all the described logic has been correctly implemented.
